tests/Doctrine/Tests/DBAL/Schema/PostgreSqlListTableForeignKeysTest.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Doctrine\Tests\DBAL\Schema;
+
+use Doctrine\DBAL\Driver\AbstractPostgreSQLDriver;
+use Doctrine\DBAL\DriverManager;
+use PHPUnit\Framework\TestCase;
+use function extension_loaded;
+use function in_array;
+
+class PostgreSqlListTableForeignKeysTest extends TestCase
+{
+    /** @var string */
+    private $testDatabaseName = 'test_pgsql_db';
+
+    /** @var string[] */
+    private $connectionParameters = [
+        'driver'   => 'pdo_pgsql',
+        'user'     => 'postgres',
+        'password' => '',
+        'host'     => 'localhost',
+        'port'     => '5432',
+    ];
+
+    protected function setUp() : void
+    {
+        if (! extension_loaded('pdo_pgsql')) {
+            $this->markTestSkipped('pdo_pgsql is not installed.');
+        } elseif (in_array('pdo_pgsql', DriverManager::getAvailableDrivers())) {
+            $connection = DriverManager::getConnection($this->connectionParameters);
+            if ($connection->getDriver() instanceof AbstractPostgreSQLDriver) {
+                $schemaManager = $connection->getSchemaManager();
+                if (in_array($this->testDatabaseName, $schemaManager->listDatabases())) {
+                    $schemaManager->dropDatabase($this->testDatabaseName);
+                }
+                $schemaManager->createDatabase($this->testDatabaseName);
+                $this->connectionParameters['dbname'] = $this->testDatabaseName;
+                $connection->close();
+                $connection = DriverManager::getConnection($this->connectionParameters);
+                $connection->executeQuery('CREATE SCHEMA test_schema1;');
+                $connection->executeQuery('CREATE SCHEMA test_schema2;');
+                $connection->executeQuery('ALTER DATABASE ' . $this->testDatabaseName . ' SET search_path TO test_schema1,test_schema2;');
+                $connection->executeQuery('CREATE TABLE test_schema1.test_foreign ( id integer PRIMARY KEY );');
+                $connection->executeQuery('CREATE TABLE test_schema1.test ( test_column varchar(5), test_foreign_id integer constraint fk_test_foreign references test_schema1.test_foreign (id) );');
+                $connection->executeQuery('CREATE TABLE test_schema2.test_foreign ( id integer PRIMARY KEY );');
+                $connection->executeQuery('CREATE TABLE test_schema2.test ( test_column varchar(5), test_foreign_id integer constraint fk_test_foreign references test_schema2.test_foreign (id) );');
+            } else {
+                $this->markTestSkipped('PostgreSQL Only test');
+            }
+        } else {
+            $this->markTestSkipped('PostgreSQL driver not available');
+        }
```

Comment:
Please use early return statements in this method rather than `else` statements.